docs/spark-ddl.md

```diff
@@ -132,12 +132,28 @@ The new table properties in the `REPLACE TABLE` command will be merged with any
 
 ## `DROP TABLE`
 
-To delete a table, run:
+The drop table behavior changed in 0.14.
+
+Prior to 0.14, running `DROP TABLE` would remove the table from the catalog and delete the table contents as well.
+
+From 0.14 onwards, `DROP TABLE` would only remove the table from the catalog.
+In order to delete the table contents `DROP TABLE PURGE` should be used.
```

Comment:
Is it worth mentioning that gc.enabled has to be true for the contents to be deleted?

Reply:
I believe it is implied that if an operation involves removing any table files then `gc.enabled` should be true.
With that said, I didn't find an occurrence of `gc.enabled` anywhere in the iceberg spark docs. Thus decided to follow the same pattern.